injection/sharedmain/main.go

```diff
@@ -266,7 +266,9 @@ func MainWithConfig(ctx context.Context, component string, cfg *rest.Config, cto
 		wh.InformersHaveSynced()
 	}
 	logger.Info("Starting controllers...")
-	go controller.StartAll(ctx, controllers...)
+	eg.Go(func() error {
+		return controller.StartAll(ctx, controllers...)
+	})
```

Comment:
note: the other changes are just a minor clean up. Running this in an error group which `Wait`s below is the fix